### PR TITLE
[cpr] Update to 1.6.2

### LINKED ISSUE #17841: [cpr] update to 1.6.2
Library name: cpr

New version number: 1.6.2

Current vcpkg support is for version 1.5.2 and latest available version is 1.6.2. https://github.com/whoshuu/cpr

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17841

Update to the latest version 1.6.2

Note: No feature needs to test.

### CODE REVIEW

File: ports/cpr/portfile.cmake [hunk 1]
```diff
@@ -1,10 +1,10 @@
-vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+#vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (NancyLi1013) [inline ports/cpr/portfile.cmake:1]:
```suggestion
vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```

**Comment by enescglyn** [comment]
Hello, how do i use this in my environment before this gets pulled?

**Comment by NancyLi1013** [comment]
> Hello, how do i use this in my environment before this gets pulled?

You can do as the following steps:

- git clone https://github.com/NancyLi1013/vcpkg.git
- cd vcpkg
- git checkout -b dev/NancyLi/update-cpr origin/dev/NancyLi/update-cpr
- .\bootstrap-vcpkg.bat
- .\vcpkg install cpr

But since there are still some problems on uwp, I'm looking into this.

**Comment by NancyLi1013** [comment]
`cpr `failed with the following errors on uwp:

```
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   ALL_SANITIZERS_AVAILABLE_EXITCODE (advanced)
   ALL_SANITIZERS_AVAILABLE_EXITCODE__TRYRUN_OUTPUT (advanced)
For details see D:/buildtrees/cpr/arm-uwp-dbg/TryRunResults.cmake
-- Performing Test ALL_SANITIZERS_AVAILABLE - Failed
```

**Comment by NancyLi1013** [comment]
Currently, add a workaround via commenting this line out to fix uwp build.
`include(cmake/sanitizer.cmake)`
https://github.com/whoshuu/cpr/blob/master/CMakeLists.txt#L49

The suggestion from https://github.com/whoshuu/cpr/issues/582#issuecomment-863054492

File: ports/cpr/portfile.cmake [hunk 2]
```diff
@@ -1,21 +1,26 @@
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
+if(vcpkg_TARGET_IS_UWP)
```
Comment (NancyLi1013) [inline ports/cpr/portfile.cmake:3]:
```suggestion
if(VCPKG_TARGET_IS_UWP)
```

File: versions/c-/cpr.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "e2cdd8ee92c53ea459a6ec6f8553051d486ff5af",
```
Comment (NancyLi1013) [inline versions/c-/cpr.json:4]:
```suggestion
      "git-tree": "96762dddaeb7d520833ddc1d47a30f26c35c9f67",
```

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**